tests/unit/s2n_crl_test.c

```diff
@@ -32,9 +32,56 @@
 #define S2N_CRL_INTERMEDIATE_INVALID_LAST_UPDATE_CRL    "../pems/crl/intermediate_invalid_last_update_crl.pem"
 #define S2N_CRL_INTERMEDIATE_INVALID_NEXT_UPDATE_CRL    "../pems/crl/intermediate_invalid_next_update_crl.pem"
 
+#define CRL_TEST_CHAIN_LEN 2
+
+struct crl_lookup_data {
+    struct s2n_crl *crls[5];
+    X509 *certs[5];
+    uint8_t callback_invoked_count;
+};
+
+static int crl_lookup_test_callback(struct s2n_crl_lookup_context *context, void *data) {
+    struct crl_lookup_data *crl_data = (struct crl_lookup_data*) data;
+
+    crl_data->callback_invoked_count += 1;
+
+    crl_data->certs[context->cert_idx] = context->cert;
+
+    struct s2n_crl *crl = crl_data->crls[context->cert_idx];
+    if (crl == NULL) {
+        s2n_crl_lookup_reject(context);
+    } else {
+        s2n_crl_lookup_accept(context, crl);
+    }
+    return 0;
+}
+
+static int crl_lookup_noop(struct s2n_crl_lookup_context *context, void *data) {
+    return 0;
+}
+
+static int crl_lookup_callback_fail(struct s2n_crl_lookup_context *context, void *data) {
+    return 1;
+}
+
+static struct s2n_crl *load_test_crl(const char* pem_path) {
+    uint8_t crl_pem[S2N_MAX_TEST_PEM_SIZE] = { 0 };
+    uint32_t pem_len = 0;
+    PTR_GUARD_POSIX(s2n_read_test_pem_and_len(pem_path, crl_pem, &pem_len, S2N_MAX_TEST_PEM_SIZE));
+    DEFER_CLEANUP(struct s2n_crl *crl = s2n_crl_new(), s2n_crl_free);
+    PTR_ENSURE_REF(crl);
+    PTR_GUARD_POSIX(s2n_crl_load_pem(crl, crl_pem, pem_len));
+
+    struct s2n_crl *crl_ret = crl;
+    ZERO_TO_DISABLE_DEFER_CLEANUP(crl);
+
+    return crl_ret;
+}
+
 int main(int argc, char *argv[])
 {
     BEGIN_TEST();
+    EXPECT_SUCCESS(s2n_disable_tls13_in_test());
 
```

Comment:
We avoid this in new tests, and control the protocol version with cipher preferences like customers do. Is there a reason you need it?

Reply:
I removed this. I originally needed it because of the old `s2n_test_cert_chain_and_key_new` function requiring the protocol version as an input. But this is no longer the case!